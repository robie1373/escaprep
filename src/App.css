.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.grid {
  position: relative;
}
.item {
  display: block;
  position: absolute;
  width: 100px;
  height: 100px;
  margin: 5px;
  z-index: 1;
  background: #112;
  color: #fff;
}
.item.muuri-item-dragging {
  z-index: 3;
}
.item.muuri-item-releasing {
  z-index: 2;
}
.item.muuri-item-hidden {
  z-index: 0;
}
.item-content {
  position: relative;
  width: 100%;
  height: 100%;
}

.left-arrow {
  padding-right: 1em;
}

.right-arrow {
  padding-left: 1em;
}

.single-line {
  white-space: nowrap;
}

.single-line-element {
  display: inline-block;
}

.single-line-img {
  display: inline-block;
  max-width: 2em;
}

#appDescription {
  max-width: 40%;
}

#react-download-form {
  max-width: 40%;
}

.react-file-reader {
  max-width: 40%;
}

.csv-table {
  margin-top: 4em;
  border-radius: 1em;
  /*
  margin-left: 1em;
  margin-right: 1em;
  border: 1px solid gray;
  */
}
