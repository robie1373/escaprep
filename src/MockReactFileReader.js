import React, { Component } from 'react';

class MockReactFileReader extends Component {
  render () {
    return (
      <div
         className="react-file-reader"
       >
         <input
           accept=".csv"
           className="react-file-reader-input"
           id="0d8f72bc-f98a-415e-aba9-c2b77dfc243c"
           multiple={false}
           onChange={[Function]}
           type="file"
         />
         <div
           className="react-file-reader-button"
           onClick={[Function]}
         >
           <button
             className="fileUploadButton"
           >
             Upload File
           </button>
         </div>
       </div>
    );
  }
}

export default MockReactFileReader;
