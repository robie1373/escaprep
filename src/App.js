import React, { Component } from 'react';
import './App.css';
import ShowHeader from './ShowHeader';
import DescribeApp from './DescribeApp';
import FileForm from './FileForm';
import DisplayTable from './DisplayTable';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      //parsedData: {data: [{"uno": 1, "two": 2, "three": 3, rowNumber: 1}]},
      //headers: ["uno", "two", "three"]
    }

  }

  componentDidMount() {
    //this.getHeaders();
  }

  stateSetter = data => {
    console.log(data.data);
    this.getHeaders(data);
    for (var i=0; i<data.data.length; i++) {
      console.log(data.data[i]);
      data.data[i].rowNumber = (i + 1);
    }
    this.setState({parsedData: data});
    console.log(this.state.parsedData);
  }

  getHeaders = (data) => {
    const parsedHeaders = Object.keys(data.data[0]);
    console.log(parsedHeaders);
    this.setState({headers: parsedHeaders});
  }

  testableFileForm = () => {
    if (this.props.filereader) {
      return(
        <FileForm filereader={this.props.filereader}
          parsedData={this.state.parsedData}
          stateSetter={this.stateSetter}
        />
      );
    }
    else {
      return(
        <FileForm parsedData={this.state.parsedData}
          stateSetter={this.stateSetter}
        />
      );
    }
  }

  toBeRendered = () => {
    if (this.state.parsedData) {
      return (
        <div>
          <ShowHeader />
          <DescribeApp />
          <this.testableFileForm />
          <DisplayTable headers={this.state.headers} data={this.state.parsedData}/>
        </div>
      );
    } else {
      return (
        <div>
          <ShowHeader />
          <DescribeApp />
          <this.testableFileForm />
        </div>
      );
    }
  }

  render() {
    return (
      <div className="App">
        <this.toBeRendered />
      </div>
    );
  }
}
/*
<ShowHeader />
<DescribeApp />
<this.testableFileForm />
<DisplayTable headers={this.state.headers} data={this.state.parsedData}/>
*/
export default App;
