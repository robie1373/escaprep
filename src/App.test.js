import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import MockReactFileReader from './MockReactFileReader';

test('snapshot', () => {
  const component = renderer.create(<App filereader={MockReactFileReader }/>);

  let tree = component.toJSON()
  ;
  expect(tree).toMatchSnapshot();
});
