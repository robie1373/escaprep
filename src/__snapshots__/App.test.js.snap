// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="App"
>
  <div>
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Escalation Preparation
      </h1>
    </header>
    <div
      className="appDescription"
    >
      <p>
        This page is used to import a single csv file from Phishme, rearrange the columns to support the powershell script and allow the user to download the new file.
      </p>
    </div>
    <div
      className="fileForm"
    >
      <div
        className="react-file-reader"
      >
        <input
          accept=".csv"
          className="react-file-reader-input"
          id="0d8f72bc-f98a-415e-aba9-c2b77dfc243c"
          multiple={false}
          onChange={
            Array [
              [Function],
            ]
          }
          type="file"
        />
        <div
          className="react-file-reader-button"
          onClick={
            Array [
              [Function],
            ]
          }
        >
          <button
            className="fileUploadButton"
          >
            Upload File
          </button>
        </div>
      </div>
      <pre>
        
      </pre>
      <pre />
    </div>
  </div>
</div>
`;
